pkg/pillar/cmd/zedmanager/updatestatus.go

```diff
@@ -577,6 +578,30 @@ func doActivate(ctx *zedmanagerContext, uuidStr string,
 	if !status.ActivateInprogress && !status.Activated &&
 		!ctx.globalConfig.GlobalValueBool(types.IgnoreMemoryCheckForApps) {
 
+		// If we have not yet calculated memory overhead - do it now
+		if status.MemOverhead == 0 {
+			// Get hypervisor
+			hyp, err := hypervisor.GetHypervisor(*ctx.hypervisorPtr)
+			if err != nil {
+				log.Fatalf("Cannot get hypervisor: %s", err)
+			}
+
+			// Create a mock domain config to calculate memory overhead
+			mockDomainConfig := types.DomainConfig{
```

Comment:
lookupDomainConfig() instead ?

Reply:
The problem is that it does not yet exist for the VM that we are creating. We create the config later in this function on MayebeAddDomainConfig (or something similar) only after we are sure we have enough recourses.

Reply:
You should be able to have `AppInstanceConfig` in you hands received from the `lookupAppInstanceConfig`. (the config should be completely parsed to this point): so `AppInstanceConfig.IoAdapterList` and other fixed resourcs. So probably to make the function cleaner (not mocking fake domain config) makes sense to adapt the `hypervisor.CountMemOverhead()` to the set of parameters and not a complete domain config. Just thoughts.

Reply:
Yeah, sure. I can also introduce a new structure for passing this args. I just didn't want to change the signatures of the original functions used to estimate the overhead (they are using exactly DomainConfig). But maybe it is worth it.

Reply:
I've switched to explicit parameters instead of DomainConfig.